src/Stratis.Features.SQLiteWalletRepository/DBConnection.cs

```diff
@@ -536,11 +536,13 @@ internal IEnumerable<(string txId, long creationTime)> RemoveAllUnconfirmedTrans
             string outputFilter = $@"
             WHERE   OutputBlockHeight IS NULL
             AND     OutputBlockHash IS NULL
+            AND     OutputTxId IS NOT NULL
             AND     WalletId = {strWalletId}";
 
             string spendFilter = $@"
             WHERE   SpendBlockHeight IS NULL
             AND     SpendBlockHash IS NULL
+            AND     SpendTxId IS NOT NULL
```

Comment:
What are these lines for? When would SpendTxId be null in the SpendingDetails table?

Reply:
The `HDTransactionData` table has two "components"; the OutputXXX fields which are associated with UTXOs present in the wallet, and the SpendXXX fields which provide information about which transaction spent the UTXO (if any). When selecting spends it's important to check that the spend actually exists to avoid doing unnecessary work.